src/js/UnifileService.js

```diff
@@ -154,11 +154,17 @@ export default class UnifileService {
     });
   }
 
+  rejectError (reject, error) {
+    reject(new Error(JSON.parse(error.data).message));
+  }
+
   // The auth method has to be called on a click or keydown in order not to be blocked by the browser
   auth (service) {
     return new Promise((resolve, reject) => {
       // Open a blank window right away, before we know the URL, otherwise the browser blocks it
       const win = window.open();
+      window.addEventListener('message', (error) => this.rejectError(reject, error));
```

Comment:
if this happens, "startPollingAuthWin" will still call "resolve"

Reply:
Holly shit, I've just forgot the return word !